Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -1678,55 +1681,105 @@ void HTMLMediaElement::loadResource(const URL& initialURL, const ContentType& co
 
     updateVolume();
 
-    bool loadAttempted = false;
-#if ENABLE(MEDIA_SOURCE)
-    if (!m_mediaSource && url.protocolIs(mediaSourceBlobProtocol) && !m_remotePlaybackConfiguration) {
-        if (RefPtr mediaSource = MediaSource::lookup(url.string()))
-            m_mediaSource = MediaSourceInterfaceMainThread::create(mediaSource.releaseNonNull());
+    auto contentType = initialContentType;
+
+    if (m_blob && !m_remotePlaybackConfiguration) {
+        ALWAYS_LOG(LOGIDENTIFIER, "loading generic blob");
+        if (!m_blobURLForReading.isEmpty())
+            ThreadableBlobRegistry::unregisterBlobURL(m_blobURLForReading);
+        m_blobURLForReading = { BlobURL::createPublicURL(document().protectedSecurityOrigin().ptr()), document().topOrigin().data() };
+        ThreadableBlobRegistry::registerBlobURL(document().protectedSecurityOrigin().ptr(), document().policyContainer(), m_blobURLForReading, m_blob->url());
+
+        url = m_blobURLForReading;
+        if (contentType.isEmpty())
+            contentType = ContentType { m_blob->type() };
     }
 
-    if (m_mediaSource) {
-        loadAttempted = true;
+    auto completionHandler = [url, player = m_player, keySystem, this](const ContentType& contentType) {
+#if ENABLE(MEDIA_SOURCE)
+        if (!m_mediaSource && url.protocolIs(mediaSourceBlobProtocol) && !m_remotePlaybackConfiguration) {
+            if (RefPtr mediaSource = MediaSource::lookup(url.string()))
+                m_mediaSource = MediaSourceInterfaceMainThread::create(mediaSource.releaseNonNull());
+        }
 
-        ALWAYS_LOG(LOGIDENTIFIER, "loading MSE blob");
-        if (!m_mediaSource->attachToElement(WeakPtr { *this })) {
-            // Forget our reference to the MediaSource, so we leave it alone
-            // while processing remainder of load failure.
-            m_mediaSource = nullptr;
-            mediaLoadingFailed(MediaPlayer::NetworkState::FormatError);
-        } else if (RefPtr mediaSource = m_mediaSource; !mediaSource->client() || !player->load(url, contentType, *mediaSource->client())) {
-            // We have to detach the MediaSource before we forget the reference to it.
-            mediaSource->detachFromElement();
-            m_mediaSource = nullptr;
-            mediaLoadingFailed(MediaPlayer::NetworkState::FormatError);
+        if (m_mediaSource) {
+            ALWAYS_LOG(LOGIDENTIFIER, "loading MSE blob");
+            if (!m_mediaSource->attachToElement(WeakPtr { *this })) {
```

Comment:
WeakPtr { }` is unnecessary.

Reply:
`virtual bool attachToElement(WeakPtr<HTMLMediaElement>&&) = 0;`

Reply:
Yes, but my comment still stands, you don't need the explicit `WeakPtr { }` as far as I know and you can just pass `*this`.